sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/GaugeResult.java

```diff
@@ -18,11 +18,14 @@
 package org.apache.beam.sdk.metrics;
 
 import com.google.auto.value.AutoValue;
+import org.apache.beam.sdk.annotations.Experimental;
+import org.apache.beam.sdk.annotations.Experimental.Kind;
 import org.joda.time.Instant;
 
 /**
  * The result of a {@link Gauge} metric.
  */
+@Experimental(Kind.METRICS)
```

Comment:
Both EmptyGaugeResult constructor and EmptyGaugeResult INSTANCE field are private. This could only be used as a return type or in an instanceof/cast. I agree it should probably be private, but that should probably wait for a future PR.